avocado/core/mux.py

```diff
@@ -163,15 +164,25 @@ def initialize_mux(self, root, mux_path, debug):
         self.root = root
         self.mux_path = mux_path
         self.debug = debug
+        self.variant_ids = self._get_variant_ids()
+
+    def _get_variant_ids(self):
+        variant_ids = []
+        for variant in MuxTree(self.root):
+            variant.sort(key=lambda x: x.path)
+            fingerprint = "-".join(_.fingerprint() for _ in variant)
+            variant_ids.append("-".join(node.name for node in variant) + '-' +
+                               sha.sha(fingerprint).hexdigest()[:4])
```

Comment:
> The hash is quite important as we'll be using it (along with the name) as a base for the per-variant-data-dir and the content of such data-dir could depend not just on the names, but also on the values.

Ok, so, it will not only be useless from the variant-ID uniqueness perspective but it will also make it harder to use the 'output check' . Every time one value changes in the yaml (or a filter. or anything), users will have to execute the `--output-check-record` to record the `std[err|out].expected` into the new directory created out off the variant-ID. Is that really what you want?
I'm not convinced we should relate the variant-ID to the values/filters/everything inside the yaml file with the only purpose of creating different .data directories names for different variant fingerprints. Could you demonstrate a use case where this is really important? I see only disadvantages so far. Please enlighten me, if you don't mind.